### DISCUSSION TITLE
海思采集视频推送到RTSP server

### DISCUSSION THREAD

**Discussion by AcSully** [ireader/media-server#220] [Q&A] [answered]
最近在看media-server源码，没有找到关于开发板推流的相关代码，请问如何去实现这个呢

**Comment by ireader** [comment, marked as answer]
可以调用rtsp-client announced & record 接口推流

**Reply by Dw9** [reply]
我们用陈老师的库和海思的mpp，实现了rtsp/rtmp/国标ps的客户端，非常nice👍

**Reply by AcSully** [reply]
有demo代码么

**Reply by ireader** [reply]
之前也有人问来着， 周末我更新一个rtsp推流的demo吧

**Reply by Dw9** [reply]
> 有demo代码么

目前主要是参考librtsp/librtmp 下面client的相关代码，需要一些二次开发。

等着后面大家和陈老师一起完善更多开箱即用的demo和文档，使用起来就会越来越爽。😁

**Reply by AcSully** [reply]
> 之前也有人问来着， 周末我更新一个rtsp推流的demo吧

good good

**Comment by ireader** [comment]
写了个简单的推流demo，librtsp/test/rtsp-client-push-test.cpp， 供参考

**Reply by AcSully** [reply]
好的，收到，谢谢你
